File: scala/4.scala
def isPalindrome(n: Int) = n.toString.reverse.toInt == n
val xs = for (
  i <- 999 to 1 by -1;
  j <- i to 1 by -1;
  if (isPalindrome(i * j))
) yield i * j

println(xs.reduceLeft(_ max _))

File: javascript/4.js
String.prototype.reverse = function () {
  return Array.prototype.slice.apply(this).reverse().join('');
};

String.prototype.isPalindrome = function () {
  // Note that a primitive JavaScript string is not === to a String object.
  // The valueOf function returns the underlying primitive.
  return this.valueOf() === this.reverse().valueOf();
};

let result = 0;
for (let i = 999; i > 0; i--) {
  for (let j = i; j > 0; j--) {
    let x = i * j;
    if (x.toString().isPalindrome() && x > result)
      result = x;
  }
}
print(result);
